packages/grid/x-data-grid/src/hooks/features/virtualization/useGridVirtualScroller.tsx

```diff
@@ -147,6 +147,12 @@ export const useGridVirtualScroller = (props: UseGridVirtualScrollerProps) => {
     height: null,
   });
   const prevTotalWidth = React.useRef(columnsTotalWidth);
+  // Each visible row (not to be confused with a filter result) is composed of a central row element
+  // and up to two additional row elements for pinned columns (left and right).
+  // When hovering any of these elements, the :hover styles are applied only to the row element that
+  // was actually hovered, not its additional siblings. To make it look like a contiguous row,
+  // we add/remove the .Mui-hovered class to all of the row elements inside one visible row.
+  const [hoveredRowId, setHoveredRowId] = React.useState<string | null>(null);
```

Comment:
I believe this is the proper way to handle the hover state, although this might have performance implications.
@romgrk What do you think?

Reply:
So IIRC, the row component is memoized so it won't re-render unless its props change. The `hovered` prop is the correct technique because the boolean value will only change for the rows where the hover state actually changed, whereas passing `hoveredRowId` down would be incorrect.

The only adjustment would be to wrap the `onMouse(Enter|Leave)` handlers with `useEventCallback` to prevent those props from changing.

Reply:
I couldn't use hooks there because they're defined in the `getRows` function.
I used our own `rowMouseEnter` and `rowMouseLeave` events instead of overriding `onMouseEnter` and `inMouseLeave` props.